lib/mllp/client.ex

```diff
@@ -34,6 +40,67 @@ defmodule MLLP.ClientContract do
 end
 
 defmodule MLLP.Client do
+  @moduledoc """
+  MLLP.Client provides a simple tcp client for sending and receiving data i
+  via [MLLP](https://www.hl7.org/documentcenter/public/wg/inm/mllp_transport_specification.PDF) over TCP.
+
+  While MLLP is primarily used to send [HL7](https://en.wikipedia.org/wiki/Health_Level_7) messages, 
+  MLLP.Client can be used to send non-hl7 messages, such as XML.
+
+  ## Connection Behaviour 
+
+  Upon successful start up via `start_link/4`, the  client will attempt to establish a connection to the given address 
+  on the provided port. If a connection can not be immediately established, the client will keep 
+  trying to establish a connection per the value of `:auto_reconnect_interval` which defaults to 
+  1 second. Therefor it is possible that before a connection is fully established, the caller
+  may attempt to send a message which will result in `MLLP.Client.Error.t()` being returned containing
+  the last error encountered in trying to establish a connection. Additionally, said behavour could be encountered
+  at any point during life span of an MLLP.Client process if the connection becomees severed on either side. 
+
+  All connection, send, and receive failures will be logged as errors.
+
+  ## Examples
```

Comment:
I think we should start a receiver so examples can work or point it to `MLLP.Receiver` module to start receiver.

Reply:
Done.